bridge-history-api/internal/controller/history_controller.go

```diff
@@ -63,10 +103,66 @@ func (c *HistoryController) PostQueryTxsByHash(ctx *gin.Context) {
 		types.RenderFailure(ctx, types.ErrParameterInvalidNo, err)
 		return
 	}
-	result, err := c.historyLogic.GetTxsByHashes(ctx, req.Txs)
-	if err != nil {
-		types.RenderFailure(ctx, types.ErrGetTxsByHashFailure, err)
+
+	if len(req.Txs) > 10 {
+		types.RenderFailure(ctx, types.ErrParameterInvalidNo, errors.New("the number of hashes in the request exceeds the allowed maximum"))
 		return
 	}
-	types.RenderSuccess(ctx, &types.ResultData{Result: result, Total: 0})
+
+	// deduplicate
+	hashesMap := make(map[string]struct{})
+	for _, hash := range req.Txs {
+		hashesMap[hash] = struct{}{}
+	}
+
+	txHashes := make([]string, 0, len(hashesMap))
+	for hash := range hashesMap {
```

Comment:
Don't need to use this loop, filter tx_hash when `range req.Txs` can be more simple.

Reply:
changed in [8b93d0e](https://github.com/scroll-tech/scroll/pull/966/commits/8b93d0ea762262922636710ded81bd8fa653f5da)